soda_machine.rb

```diff
@@ -7,12 +9,20 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    sodas.length
```

Comment:
[getters and setters are for reading data OUTSIDE the class](https://railsadventures.wordpress.com/2014/12/10/misusing-rubys-attribute-readers-and-writers/). Since you are still inside the SodaMachine class and developing an instance method, you should be using the instance variable.